src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -246,7 +246,21 @@ public final void setTabWidth(int tabWidth) {
      * @param args arguments to format
      */
     public final void log(DetailAST ast, String key, Object... args) {
-        log(ast.getLineNo(), ast.getColumnNo(), key, args);
+        final int col = 1 + CommonUtils.lengthExpandedTabs(
+                getLines()[ast.getLineNo() - 1], ast.getColumnNo(), tabWidth);
```

Comment:
please explain this +1 and -1 in comment. It looks scary without comment.

Reply:
done